### PR TITLE
Begin matching ft_unkown_006.c and mncharsel.c

### PR DESCRIPTION
**Pull request by Tri-Wing**

### CODE REVIEW

File: src/melee/ft/ft_unknown_006.c [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+extern s32 func_800233EC(s32);
+extern s32 func_800C06B4(struct _Fighter*);
+
+s32 func_800877F8(HSD_GObj* fighterObj, s32 arg1)
+{
+    Fighter* fp = getFighter(fighterObj);
+    if (fp->input.x65C_heldInputs & arg1) {
+        return 1;
+    } else {
+        return 0;
```
Comment (ribbanya) [inline src/melee/ft/ft_unknown_006.c:19]: Would prefer to see ternary operators for these with `TRUE` and `FALSE`, or a macro to take care of it.

File: src/melee/ft/ft_unknown_006.c [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+    f32 atk_shield_kb_x;
+    f32* atk_shield_kb_y;
+    f32 fVar1;
+    f32 fVar2;
+    Fighter* fp;
+    fp = getFighter(fighter_gobj);
+    atk_shield_kb_x = fp->x98_atk_shield_kb.x;
+    fVar1 = (fp->x80_self_vel.x * fp->x80_self_vel.x) + (fp->x80_self_vel.y * fp->x80_self_vel.y);
+    atk_shield_kb_y = &fp->x98_atk_shield_kb.y;
+    fVar2 = (atk_shield_kb_x * atk_shield_kb_x) + ((*atk_shield_kb_y) * (*atk_shield_kb_y));
```
Comment (ribbanya) [inline src/melee/ft/ft_unknown_006.c:201]: Can combine the declarations with the initializers here.

File: src/melee/ft/ft_unknown_006.c [hunk 3]
```diff
@@ -0,0 +1,718 @@
...
+    var0 = fp->x1A98;
+    func_800A101C(fp, arg1, var0, fp->x1A9C);
+}
+
+void func_80087BEC(HSD_GObj* fighterObj, u8 arg1)
+{
+    Fighter* fp;
+    u64 var0;
+    fp = getFighter(fighterObj);
+    var0 = fp->x1A94;
```
Comment (ribbanya) [inline src/melee/ft/ft_unknown_006.c:234]: Combine here, too.

File: src/melee/ft/ft_unknown_006.c [hunk 4]
```diff
@@ -0,0 +1,718 @@
...
+                {
+                    sfx_id += 0x66;
+                }
+                if (((0x1FC63 <= sfx_id) && (sfx_id <= 0x1FCC8)) && (func_80080144(fighter) != 1))
+                {
+                    sfx_id -= 0x66;
+                }
+            }
+        }
+        */
```
Comment (ribbanya) [inline src/melee/ft/ft_unknown_006.c:406]: Does this still serve a purpose?

File: src/melee/ft/ft_unknown_006.c [hunk 5]
```diff
@@ -0,0 +1,718 @@
...
+/* 80088048 00084C28  83 C1 00 28 */	lwz r30, 0x28(r1)
+/* 8008804C 00084C2C  83 A1 00 24 */	lwz r29, 0x24(r1)
+/* 80088050 00084C30  38 21 00 30 */	addi r1, r1, 0x30
+/* 80088054 00084C34  7C 08 03 A6 */	mtlr r0
+/* 80088058 00084C38  4E 80 00 20 */	blr 
+}
+#endif
+
+#pragma push
+#pragma peephole on
```
Comment (ribbanya) [inline src/melee/ft/ft_unknown_006.c:688]:
```suggestion
#pragma pop
```

File: src/melee/ft/ft_unknown_006.c [hunk 6]
```diff
@@ -0,0 +1,718 @@
...
+        }
+        break;
+    default:
+        break;
+    }
+
+    return sfx_id;
+}
+#else
+asm s32 func_80087D0C(Fighter* fighter, s32 arg1)
```
Comment (ribbanya) [inline src/melee/ft/ft_unknown_006.c:444]:
```suggestion

#pragma push

asm s32 func_80087D0C(Fighter* fighter, s32 arg1)
```

File: src/melee/ft/ft_unknown_006.c [hunk 7]
```diff
@@ -0,0 +1,718 @@
...
+{
+    func_8002500C(fp->x2164);
+    fp->x2164 = 0;
+}
+
+void func_80088110(Fighter* fp)
+{
+    func_80025038(fp->x2168);
+    fp->x2168 = 0;
+}
```
Comment (ribbanya) [inline src/melee/ft/ft_unknown_006.c:718]:
```suggestion
}

```

File: src/melee/ft/ft_unknown_006.h [hunk 8]
```diff
@@ -0,0 +1,46 @@
+#ifndef _ftunk006_h_
+#define _ftunk006_h_
+
+#include <global.h>
```
Comment (ribbanya) [inline src/melee/ft/ft_unknown_006.h:4]: Is this used anywhere?

File: src/melee/mn/mncharsel.c [hunk 9]
```diff
@@ -0,0 +1,123 @@
+
+#include "mncharsel.h"
+#include "mntext.h"
```
Comment (ribbanya) [inline src/melee/mn/mncharsel.c:3]: Use `<full/path/header.h>`.

File: src/melee/mn/mncharsel.h [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+typedef struct _CSSData {
+    u8 unk_0x0;
+    u8 unk_0x1;
+    u8 match_type;
+    u8 pending_scene_change;
+    u8* ko_star_counts;
+    VsModeData data;
+} CSSData;
+
+#endif
```
Comment (ribbanya) [inline src/melee/mn/mncharsel.h:173]:
```suggestion
#endif

```

File: src/melee/mn/mntext.h [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    /*0x00*/ u8 left;
+    /*0x01*/ u8 right;
+} TextKerning;
+
+typedef struct _SIS {
+    /*0x00*/ TextKerning* kerning;
+    /*0x04*/ TextGlyphTexture* textures;
+} SIS;
+
+#endif
```
Comment (ribbanya) [inline src/melee/mn/mntext.h:22]:
```suggestion
#endif

```

**Review by ribbanya — CHANGES_REQUESTED** [review]

File: src/melee/mn/mncharsel.c [hunk 12]
```diff
@@ -1,6 +1,6 @@
 
-#include "mncharsel.h"
-#include "mntext.h"
+#include <src/melee/mn/mncharsel.h>
+#include <src/melee/mn/mntext.h>
```
Comment (ribbanya) [inline src/melee/mn/mncharsel.c:3]:
```suggestion
#include <melee/mn/mncharsel.h>
#include <melee/mn/mntext.h>
```

**Comment by Tri-Wing** [comment]
Unsure why checks are failing, says the #pragma push is missing but it is there

**Review by ribbanya — APPROVED** [review]
Looks good now. Thanks for the PR!

**Pull request merged**